pkg/monitor-hw/cmd/dell.go

```diff
@@ -2,46 +2,15 @@ package cmd
 
 import (
 	"context"
-	"os"
-	"time"
 
-	"github.com/cybozu-go/log"
 	"github.com/cybozu-go/well"
 )
 
 func monitorDell(ctx context.Context) error {
 	if err := initDell(ctx); err != nil {
 		return err
 	}
-	if err := resetDell(ctx); err != nil {
-		return err
-	}
-
-	env := well.NewEnvironment(ctx)
-	env.Go(func(ctx context.Context) error {
-		for {
-			select {
-			case <-time.After(time.Duration(opts.resetInterval) * time.Hour):
-			case <-ctx.Done():
-				return nil
-			}
-
-			if _, err := os.Stat(opts.noResetFile); err == nil {
-				// if no-reset file exists, skip reset.
-				continue
-			}
-
-			if err := resetDell(ctx); err != nil {
-				log.Error("failed to reset iDRAC", map[string]interface{}{
-					log.FnError: err,
-				})
-				// continue working
-			}
-		}
-	})
-
-	env.Stop()
-	return env.Wait()
+	return resetDell(ctx)
```

Comment:
Please refactor the caller of this facntion.
There is no longer a need to run this function as a goroutine.
https://github.com/cybozu-go/setup-hw/blob/v1.15.1/pkg/monitor-hw/cmd/root.go#L107-L109

Reply:
I refactored it by deleting monitor functions. Also, I changed `startExporter()` to keep running without `well.Wait()`.

Reply:
Please check whether monitor-hw exits normally.
As far as I have confirmed, the following test programs did not terminate when receiving signals.
I think monitor-hw will be the same.

```go
package main

import (
	"context"
	"io"
	"net/http"
	"time"

	"github.com/cybozu-go/log"
	"github.com/cybozu-go/well"
	"github.com/spf13/cobra"
)

func startExporter() error {
	well.Go(func(ctx context.Context) error {
		for {
			log.Info("Do collector.Update(ctx)", nil)
			select {
			case <-time.After(time.Second):
			case <-ctx.Done():
				return nil
			}
		}
	})

	mux := http.NewServeMux()
	mux.HandleFunc("/metrics", func(w http.ResponseWriter, r *http.Request) {
		io.WriteString(w, "Hello")
	})
	return http.ListenAndServe(":9105", mux)
}

var rootCmd = &cobra.Command{
	PersistentPreRun: func(cmd *cobra.Command, args []string) {
		cmd.SilenceUsage = true
		err := well.LogConfig{}.Apply()
		if err != nil {
			log.ErrorExit(err)
		}
	},

	RunE: func(cmd *cobra.Command, args []string) error {
		return startExporter()
	},
}

func main() {
	err := rootCmd.Execute()
	if err != nil {
		log.ErrorExit(err)
	}
}
```

Reply:
Thank you.
I rollbacked to use well.HTTPServer.ListenAndServe() and call `well.Stop()` and `well.Wait` at the parent function.